calus.js

```diff
@@ -1,228 +1,264 @@
+            if (typeof array[0] == "string") {
+                // check that they are valid ISO dates;
+                let ISODates = [];
 
-                    this.onSelect(day)
+                try {
+                    ISODates = array.map((x) => {
+                        let isoDate = DateTime.fromISO(x);
+                        if (isoDate.isValid) {
+                            return isoDate;
+                        } else {
+                            throw "Invalid ISO string found";
+                        }
+                    });
+                } catch (error) {
+                    console.error(error);
                 }
-            },
-            setAvailable: function (array) {
-                this.resetSelected()
-
-                if (typeof array[0] == 'string') {
-
-                    // check that they are valid ISO dates
-                    let ISODates = []
-
-                    try {
-                        ISODates = array.map((x) => {
-                            let isoDate =  DateTime.fromISO(x)
-                            if (isoDate.isValid) {
-                                return isoDate
-                            } else {
-                                throw "Invalid ISO string found"
-                            }
-                        })
-                    } catch (error) {
-                        console.error(error)
-                    }
 
-                    this.availableDates = ISODates
+                this.datesAvailable = ISODates;
+            } else {
+                this.datesAvailable = array;
+            }
+        },
+        // resets the selected day
+        resetSelected: function () {
+            this.selectedDate = null;
+            this.removeSelectedStyle();
+        },
+        addSelectedStyle: function (time) {
+            let selectedEl = this.$el.querySelector(
+                '[data-date="' + +time + '"]'
+            );
+            if (selectedEl) {
+                selectedEl.classList.add("is-selected");
+            }
+        },
+        removeSelectedStyle: function () {
+            let currentSelectedDay = this.$el.querySelector(".day.is-selected");
+            if (currentSelectedDay) {
+                currentSelectedDay.classList.remove("is-selected");
+            }
+        },
+        scrollMonth: function (delta) {
+            if (!this.displayInColumn) {
+                this.removeSelectedStyle();
+                this.currentDisplayedMonth = this.currentDisplayedMonth.plus({
+                    months: delta,
+                });
+                setTimeout(() => {
+                    this.addSelectedStyle(this.selectedDate);
+                }, 1);
+            }
+        },
+        disableScroll: function () {
+            const min = this.firstAvailable;
+            const max = this.lastAvailable;
+            const prev = this.monthControlsClasses[0];
+            const next = this.monthControlsClasses[1];
```

Comment:
The default in `props` is an empty array so going to get an undefined in `prev`, `next` which you're then using as a `querySelector()`

Reply:
this function - disableScroll() will only be called if `this.monthControlsClasses` isn't empty so will still work